app/helpers/post_location_helper.rb

```diff
@@ -0,0 +1,17 @@
+module PostLocationHelper
+  def link_to_post(post)
+    if post.type == "ExternalPost"
+      link_to(post.title, post.source_url, target: '_blank')
+    else
+      link_to(post.title, post_path(post))
+    end
+  end
+
+  def show_tags(post)
+    if post.type == "ExternalPost"
+      "<span>question</span>".html_safe
```

Comment:
Instead of hard-coding the tag in the HTML, the tag should be added when the record is created. There should be a `before_create` callback which sets `tags_string = "question"`